tests/certification/secretstores/hashicorp/vault/vault_test.go

```diff
@@ -0,0 +1,238 @@
+		assert.Error(t, getErr)
+
+		return nil
+	}
+
+	flow.New(t, "Test component is up and we can retrieve some secrets").
+		Step(dockercompose.Run(dockerComposeProjectName, dockerComposeClusterYAML)).
+		Step("Waiting for component to start...", flow.Sleep(5*time.Second)).
+		Step(sidecar.Run(sidecarName,
+			embedded.WithoutApp(),
+			embedded.WithComponentsPath(secretStoreComponentPath),
+			embedded.WithDaprGRPCPort(currentGrpcPort),
+			embedded.WithDaprHTTPPort(currentHttpPort),
+			componentRuntimeOptions(),
+		)).
+		Step("Waiting for component to load...", flow.Sleep(5*time.Second)).
+		Step("Verify component is registered", testComponentFound(t, secretStoreName, currentGrpcPort)).
+		Step("Run basic secret retrieval test", testGetKnownSecret).
+		Step("Test retrieval of secret that does not exist", testGetMissingSecret).
+		Step("Interrupt network for 1 minute",
+			network.InterruptNetwork(networkInstabilityTime, nil, nil, servicePortToInterrupt)).
+		Step("Wait for component to recover", flow.Sleep(waitAfterInstabilityTime)).
+		Step("Run basic test again to verify reconnection occurred", testGetKnownSecret).
+		Step("Stop HashiCorp Vault server", dockercompose.Stop(dockerComposeProjectName, dockerComposeClusterYAML)).
+		Run()
+}
+
+func TestMultipleKVRetrieval(t *testing.T) {
+	ports, err := dapr_testing.GetFreePorts(2)
+	assert.NoError(t, err)
+
+	currentGrpcPort := ports[0]
+	currentHttpPort := ports[1]
+
+	testGetMultipleKeyValuesFromSecret := func(ctx flow.Context) error {
+		client, err := client.NewClientWithPort(fmt.Sprint(currentGrpcPort))
+		if err != nil {
+			panic(err)
+		}
+		defer client.Close()
+
+		emptyOpt := map[string]string{}
+
+		// This test reuses the HashiCorp Vault's conformance test resources created using
+		// .github/infrastructure/docker-compose-hashicorp-vault.yml,
+		// so it reuses the tests/conformance/secretstores/secretstores.go test secrets.
+		res, err := client.GetSecret(ctx, secretStoreName, "multiplekeyvaluessecret", emptyOpt)
+		assert.NoError(t, err)
+		assert.NotNil(t, res)
+		assert.Equal(t, "1", res["first"])
+		assert.Equal(t, "2", res["second"])
+		assert.Equal(t, "3", res["third"])
+		return nil
+	}
+
+	flow.New(t, "Test retrieving multiple key values from a secret").
+		Step(dockercompose.Run(dockerComposeProjectName, dockerComposeClusterYAML)).
+		Step("Waiting for component to start...", flow.Sleep(5*time.Second)).
+		Step(sidecar.Run(sidecarName,
+			embedded.WithoutApp(),
+			embedded.WithComponentsPath(secretStoreComponentPath),
+			embedded.WithDaprGRPCPort(currentGrpcPort),
+			embedded.WithDaprHTTPPort(currentHttpPort),
+			componentRuntimeOptions(),
+		)).
+		Step("Waiting for component to load...", flow.Sleep(5*time.Second)).
+		Step("Verify component is registered", testComponentFound(t, secretStoreName, currentGrpcPort)).
+		Step("Verify component has support for multiple key-values under the same secret",
```

Comment:
FYI: the functions referenced here could be used in other tests as-is. And TBH, checking that the component  we _think_ we registered is indeed loaded is a good sanity check in those tests.